helix-term/src/commands.rs

```diff
@@ -3459,6 +3483,9 @@ fn open(cx: &mut Context, open: Open) {
     transaction = transaction.with_selection(Selection::new(ranges, selection.primary_index()));
 
     doc.apply(&transaction, view.id);
+
+    // Since we might have added a comment token, move to the end of the line.
+    goto_line_end_newline(cx);
```

Comment:
Rather than jumping to the end of the line we should update the selection ranges based on how much we're inserting in the block above with the "calculate new selection ranges" comment. I think the final formula there should be `pos + (i * (1 + indent_len + comment_len)) + indent_len + comment_len` where `comment_len` is

```rust
let comment_len = continue_comment_token.map(|token| token.len() + 1).unwrap_or_default();
```

(+ 1 for the extra space added).